src/main/java/gregtech/common/metatileentities/miner/MetaTileEntityLargeMiner.java

```diff
@@ -0,0 +1,617 @@
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static gregtech.api.unification.material.Materials.DrillingFluid;
+
+public class MetaTileEntityLargeMiner extends MultiblockWithDisplayBase implements Miner, IControllable, IDataInfoProvider, IFastRenderMetaTileEntity, IProgressBarMultiblock {
+
+    @Nonnull
+    public final LargeMinerType type;
+    public final int tier;
+    public final int drillingFluidConsumePerTick;
+
+    private final MultiblockMinerLogic minerLogic;
+
+    private IEnergyContainer energyContainer;
+    protected IMultipleTankHandler inputFluidInventory;
+    protected IItemHandlerModifiable outputInventory;
+
+    // last mined ore block
+    protected final MutableBlockPos lastMinedOre = new MutableBlockPos();
+    protected boolean hasLastMinedOre;
+    // number of ores processed so far
+    protected int minedOreCount;
+
+    private boolean inventoryFull;
+
+    public MetaTileEntityLargeMiner(ResourceLocation metaTileEntityId, int tier, int speed, int maxChunkDiameter,
+                                    int drillingFluidConsumePerTick, @Nonnull LargeMinerType type) {
+        super(metaTileEntityId);
+        this.type = Objects.requireNonNull(type, "type == null");
+        this.tier = tier;
+        this.drillingFluidConsumePerTick = drillingFluidConsumePerTick;
+        this.minerLogic = new MultiblockMinerLogic(this, speed, maxChunkDiameter);
+    }
+
+    @Override
+    public MetaTileEntity createMetaTileEntity(IGregTechTileEntity tileEntity) {
+        return new MetaTileEntityLargeMiner(metaTileEntityId, this.tier, this.minerLogic.getWorkFrequency(),
+                this.minerLogic.getMaximumChunkDiameter(),
+                this.drillingFluidConsumePerTick, this.type);
+    }
+
+    @Override
+    public void invalidateStructure() {
+        super.invalidateStructure();
+        this.inputFluidInventory = new FluidTankList(true);
+        this.outputInventory = new ItemStackHandler(0);
+        this.energyContainer = new EnergyContainerList(Lists.newArrayList());
+    }
+
+    @Override
+    protected void formStructure(PatternMatchContext context) {
+        super.formStructure(context);
+        this.inputFluidInventory = new FluidTankList(false, getAbilities(MultiblockAbility.IMPORT_FLUIDS));
+        this.outputInventory = new ItemHandlerList(getAbilities(MultiblockAbility.EXPORT_ITEMS));
+        this.energyContainer = new EnergyContainerList(getAbilities(MultiblockAbility.INPUT_ENERGY));
+    }
+
+    public int getEnergyTier() {
+        if (energyContainer == null) return this.tier;
+        return Math.min(this.tier + 1, Math.max(this.tier, GTUtility.getFloorTierByVoltage(energyContainer.getInputVoltage())));
+    }
+
+    @Override
+    public boolean drainMiningResources(@Nonnull MinedBlockType minedBlockType, boolean pipeExtended, boolean simulate) {
+        if (minedBlockType == MinedBlockType.NOTHING) return true;
```

Comment:
Does this mean that no energy is used in the scanning period while running a multiblock?